corehq/pillows/app_submission_tracker.py

```diff
@@ -36,21 +37,28 @@ def get_app_form_submission_tracker_pillow(pillow_id='AppFormSubmissionTrackerPi
 
 
 class AppFormSubmissionReindexDocProcessor(BaseDocProcessor):
-    def __init__(self, pillow_processor):
+    def __init__(self, pillow_processor, data_source_type, data_source_name):
         self.pillow_processor = pillow_processor
+        self.data_source_type = data_source_type
+        self.data_source_name = data_source_name
 
     def process_doc(self, doc):
-        change = self._doc_to_change(doc)
-        self.pillow_processor.process_change(None, change)
+        change = self._doc_to_change(doc, self.data_source_type, self.data_source_name)
+        try:
+            self.pillow_processor.process_change(None, change)
+        except Exception:
+            return False
+        else:
+            return True
 
     @staticmethod
-    def _doc_to_change(doc):
+    def _doc_to_change(doc, data_source_type, data_source_name):
         doc_meta = get_doc_meta_object_from_document(doc)
         change_meta = change_meta_from_doc_meta_and_document(
             doc_meta=doc_meta,
             document=doc,
-            data_source_type=None,
-            data_source_name=None,
+            data_source_type=data_source_type,
+            data_source_name=data_source_name,
```

Comment:
Why is this necessary? As far as I can see those are only required if you need to fetch the doc from the DB again which isn't the case here since the doc is already populated in the `Change`